slackify-html.js

```diff
@@ -212,16 +231,23 @@ function walk(dom, nesting) {
   if (!nesting) {
     nesting = 0;
   }
-  var out = "";
+  let out = "";
   if (dom)
     dom.forEach(function (el) {
+      let suffixSpace;
+      let prefixSpace;
       if ("text" === el.type) {
         out += el.data;
       } else if ("tag" === el.type) {
+        let i;
+        let content;
+        let contentArr;
+        let innerOutput;
+
         switch (el.name) {
           case "a":
             if (el.attribs && el.attribs.href) {
-              out += "<" + el.attribs.href + "|" + walk(el.children) + ">";
+              out += "<" + el.attribs.href + "|" + walkLink(el.children) + ">";
```

Comment:
That new function is used here since we don't want hidden tags to be treated the same where we're looking for just text.